### ISSUE TITLE
How to improve the model accuracy?

### ISSUE THREAD

**Issue by laifuchicago** [deepset-ai/FARM#481]
To author:
I have tried many language models such as Bert-base, Bert-multilingual, xlnet and XLM-R, but the problem is the Dev-F1 is still 74.8 percent . We also do the manual testing( by using haystack Elasticsearch retriever) , and it's still 71.68%. Do you have any suggestions that I can improve that one? My training data is Squad.
![accuracy improving](https://user-images.githubusercontent.com/8537280/88890502-4b532180-d274-11ea-9ea8-f45a71cf490f.PNG)

Jonathan Sung

**Comment by Timoeller** [comment]
deepset to the rescue : )
What do you mean by "We also do the manual testing( by using haystack Elasticsearch retriever)"?

To your issue:
Are you using FARM for finetuning the models? 
Maybe have a look at how EM and f1 scores develop during training. I do not see anything suspicious, apart from your learning rate being low and you only have 2 epochs. Might be it improves when training longer? Have a look at an XLM-r large that we finetuned here: https://huggingface.co/deepset/xlm-roberta-large-squad2

**Comment by laifuchicago** [comment]
To author:
We first upload our docs to ES, and then use xlm to test. The attached file is our testing result. It's only 71.68%, that's why I got confused. Thank you!
[(918Q)Analysis of TriviaQA.xlsx](https://github.com/deepset-ai/FARM/files/4999556/918Q.Analysis.of.TriviaQA.xlsx)

Jonathan Sung

**Comment by Timoeller** [comment]
Ok lets figure this out together. I need a bit more context to help you out.

1. Do you test on TriviaQA? Your file is named "(918Q)Analysis of TriviaQA.xlsx" which hints you do. TriviaQA is different from SQuAD and more difficult. Having a lower score there is not surprising.

2. About the MLflow screen shot. You are training a XLM-r large but possibly not long enough. Did you try our [XLM-r model](https://huggingface.co/deepset/xlm-roberta-large-squad2) and evaluated how it works?

3. You say you upload documents to ES. So you are working in haystack? Do you use a retriever? If you use a retriever it is also reasonable that the performance is going down.

**Comment by laifuchicago** [comment]
To Author:
1. We download the pure xlm model from huggingface( haven't tried your model) . Is that the reason that's not long enough?
2. We use Squad to train, The TriviaQA is just for inference.
3. We use Elasticsearch, because our data is very large. And also we want shorter inference time. 

Jonathan Sung

**Comment by laifuchicago** [comment]
To Author:
Now I use TriviaQA to train, but the training time is long. I set the batch size=24 , lr=1.5e-5, epoch=2 . Do you have any suggestions  about setting the hyperparameter? Thank you.
![triviatrain](https://user-images.githubusercontent.com/8537280/89151809-5ff52980-d594-11ea-9ec9-7454d20f04c5.PNG)
Jonathan Sung
